web/testing/promise-api/eventLogging.js

```diff
@@ -0,0 +1,49 @@
+function attachLoggingEvents() {
+
+  keyman.addEventListener('beforekeyboardchange', beforeChange);
+  keyman.addEventListener('keyboardchange', change);
+  keyman.addEventListener('keyboardloaded', loaded);
+  keyman.addEventListener('keyboardregistered', registered);
+}
+
+function appendLog(logText) {
+  if(this.loggingElement === undefined) {
```

Comment:
`this` is a little ambiguous here. It could be `window` or it could be `keyman.xxx` it seems because it depends on the caller. Can we just make `loggingElement` a global in this script?

Reply:
I... uh... we could.  Not usually one for unnecessary globals though... and I lifted this trick from you:

https://github.com/keymanapp/keyman/blob/bb4887387729a2ea5761e4e0d7d19c118eac6d9f/web/source/osk/oskKey.ts#L265-L295

Granted... yeah, encapsulation isn't so much a concern for the test page.

Reply:
Ah, I'd have had to do it more directly on the function to do the 'trick' right.

Reply:
:grin: Changes LGTM as encapsulation is not important here as you noted.